### DISCUSSION TITLE
SDongle Wifi - integration unavaiable

### DISCUSSION THREAD

**Discussion by CristiXP** [wlcrs/huawei_solar#549] [Connecting to the inverter]
Hello,in the past I used the internal Wifi network of the inverter and from time to time I got disconnects, or it worked perfect for weeks...

Since 3 days ago, I have installed a dongle to be able to install a second inverter in the future, but since the installation of the SDongle, I get 2-3 disconnects daily and it doesn't connect anymore until I reload the integration. I have created an automation to reload Inverter/ Powermetter/ Battery when sees one value unavailable... but can't this be fixed???

If I connect LAN directly to SDongle will it make it better?

Thank you!

**Comment by Roving-Ronin** [comment]
@CristiXP you haven't mentioned if you removed the integration, cleaned up the entities and then reinstall WLCRS?
If you used the comissioning (inbuilt) AP before you would have a 192.168.100.x (iirc) IP address, I assume you use a different IP range on your home network and the integration would need to be reinstalled.

I'm using the configuration you mention (option B) with WiFi to a sDongle plugged into the primary inverter (that has a secondary inverter and DTSU-666H linked via RS485 into it), and the WCLRS software works fine in this scenario.

As you mentioned, I would look to using the RJ45 ethernet connection of the SDongle that would rule out WiFi signal and drop out issues, after that I'd look at the HA instance and home network.

**Reply by CristiXP** [reply]
It has different IP, but my RaspPi it is in another location, so I am using port redirect and connecting to the inverter via DNShostname:502 for the inbuilt IP it was DNShostname:6607

**Reply by Roving-Ronin** [reply]
@CristiXP might be worthwhile using the IP-addres:502 to rule out any lag with DNS? And/or add it into the RPi's  hostname lookup file? Assume its a good connection between RPi and remote Inverter (low lag etc) ?

**Reply by CristiXP** [reply]
It worked for weeks in the past without me noticing any issue. Now with the dongle it seems not to reconnect anymore until I reload the integration so something had changed because of the Dongle. 

Refresh in const.py is set at 10s... 

How can I add it the lookup file... to prevent everytime a lookup for it? But I don't think this is a solution... as the RpI is in a different physical location...and dinamic IP address.

Could be an issue of the network, but the router is an AP mode connected directed to LAN to it and DHCP disabled on it. I have a separate network on it created special to connect IOT devices, and the distance between router and inverter is 10-15m in plain sight...

At this moment, I think the Dongle doesn't like something on my network... so I will try putting LAN directly to it.

**Comment by JensenNick** [comment]
I would keep Home Assistant on the same subnet as the inverter/dongle (you would need to “poke a hole” in your firewall anyway if not kept on the same subnet, so nothing gained AFAIK). I would also set a static ip for both Home Assistant/RPi and for the inverter/dongle and I would do this in the router. Lot of people are using WiFi so unless you have very heavy walls I don’t think this is your issue - and if you previously was running without dongle I don’t se any reason why the WiFi Dongle would cause any issues.

**Comment by jms92100** [comment]
hello ; thanks for your integration, i have just been installed with my SUN2000-6KTL-L1 . it uses a dongle connected to my wifi . 
i have installed your huawei solar but get stuck in the connection. 
whatever i try ipadress:502 slave id 0 or 1 nothing works . 
i tried to get a look on the huawei solar web site it shows a dongle SDongleA-05 wich is connected to my wifi.  i have no acces to the configuration tab . 
any idea ?

**Reply by JensenNick** [reply]
You should not put the port number after you ip address. Port number can also be 6607. Did you try with this? Slave id could also be 2 (if your installer started with the inverter and gave it 1). Try. Do help yourself and assign a static ip address to the dongle.

**Comment by jms92100** [comment]
thx, i tried but still nope..
do you think it is related to the tcp flag on the modbus ?

Le mer. 13 déc. 2023 à 18:35, JensenNick ***@***.***> a
écrit :

> You should not put the port number after you ip address. Port number can
> also be 6607. Did you try with this? Slave id could also be 2 (if your
> installer started with the inverter and gave it 1). Do help yourself and
> assign a static ip address to the dongle.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wlcrs/huawei_solar/discussions/549#discussioncomment-7845399>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG5WRE43CSU3F4HYFO2SMT3YJHRODAVCNFSM6AAAAAA7Z4Z4RSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TQNBVGM4TS>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Cordialement

**Reply by JensenNick** [reply]
TCP flag? Did you setup / enable modus on the inverter according ti the instructions?

**Reply by jms92100** [reply]
Nick , a bit of time passed. 
I am now able to get information from the inverter. 
Thanks to your help .
Home assistant is now able to get info and i am really glad of this integration. 
SO what happened ? 
1-I got the hands on the fusion solar application and was able to check and modify parameters. (tcp over modbus) 
2-updated the version of the integration 
3-retried , did not work. 
4-started the logs .. 
5- retried and got it finally  (i did not choose the privileges o access the data )
6- i am now playing with the power dashboard 
Many thanks again 
I will purssue when i will connect batteries.

**Comment by jms92100** [comment]
No i have no access to the setup. I m askingvthe installer

**Comment by jms92100** [comment]
the technical guy came tpoday , he has enabled the modbus over tcp . 
retried ip adress: port: 6607  and id 0 or 1 or 2 
still nothing  
Help

**Reply by JensenNick** [reply]
1) First, log onto your router and make sure your router is providing a ip address to your dongle and that it is active.
2) If not, try to restart your router.
3) Stil no active ip address? Shut down your PV (inverter etc. both on AC and DC). Restart your router. Turn your PV back on. This should hand out a new ip address to your dongle.
4) Still no ip address? You could have wifi coverage issues. There are numerous ways to go about this.
5) If you have an ip address check the connectivity to HA according the wiki. Replace 192.168.200.1 with your ip address.
![2023-12-24_07-50-20](https://github.com/wlcrs/huawei_solar/assets/149193835/6b44539d-155f-4525-be8d-db5e8b43e348)
https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter

**Comment by max10019679** [comment]
did you manage to resolve this? i have the same issue with a SUN2000-6KTL-L1 and huawei dongle connected to the router with RJ45 cable. RP4i/HA also connected to router with RJ45. I have the inverter on a static IP and i can ping the inverter fine but cannot complete the integration with either port 502/6607 and or slave ID 0,1,16?

**Reply by jms92100** [reply]
try to update the integration 
then look to the fusion solar parameters  (check tcp modbus . look to the id used) 
then insert the modbus log in the configuration file 
logger:
  default: info
  logs:
     pymodbus: debug # only include this if you're having connectivity issues
     huawei_solar: debug
     homeassistant.components.huawei_solar: debug
retry with th fixed ip , the id got from the parameters and i think i tried 502 port (i did not find the parameter to open the 6607)
if it doesnot work got the the logs you may see something there (it worked for me so i did not look to the logs)
good luck
